src/StatsdClient/Transport/NamedPipeTransport.cs

```diff
@@ -0,0 +1,101 @@
+#if NAMED_PIPE_AVAILABLE
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Threading;
+
+namespace StatsdClient.Transport
+{
+    internal class NamedPipeTransport : ITransport
+    {
+        private readonly NamedPipeClientStream _namedPipe;
+        private readonly TimeSpan _timeout;
+        private byte[] _internalbuffer = new byte[0];
+
+        // As SpinLock is a struct, if it is marked as `readonly`, each time it is used
+        // a new copy is created which leads to the error:
+        // System.Threading.SynchronizationLockException : The calling thread does not hold the lock.
+        private SpinLock _lock = new SpinLock(enableThreadOwnerTracking: true);
```

Comment:
Is a spinlock the right mechanism for synchronization here? If we have excessive spinning it might lead to higher CPU load than is desirable.